tests/test_cpu_adam/resnet_cifar10.py

```diff
@@ -0,0 +1,217 @@
+
+import torch
+import torch.distributed as dist
+import colossalai
+import argparse
+from tqdm import tqdm
+from torchvision.models import resnet50
+from colossalai.context.parallel_mode import ParallelMode
+from colossalai.zero import ShardedOptimizer
+from colossalai.logging import get_dist_logger, disable_existing_loggers
+from colossalai.core import global_context as gpc
+from colossalai.utils import get_dataloader
+from torchvision.datasets import CIFAR10
+from torchvision.transforms import transforms
+from torch.nn.parallel import DistributedDataParallel as DDP
+from colossalai.nn.optimizer import CPUAdam
+from colossalai.utils import Timer
+
+
+
+BATCH_SIZE = 64
+NUM_EPOCHS = 100
+
+torch.backends.cudnn.benchmark = False
+torch.backends.cudnn.deterministic = True
+
+def sync_param(model):
+    for param in model.parameters():
+        dist.broadcast(param, src=0, group=gpc.get_group(ParallelMode.DATA))
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--log')
+    parser.add_argument('--use_torch', action='store_true')
+    parser.add_argument('--local_rank')
+    return parser.parse_args()
```

Comment:
There is no need to add convergence test to unit testing. Unit testing is at a finer granularity. You can just compare this implementation with normal pytorch adam.

Reply:
Thanks. I've removed the convergence test.